source/Stargate-Examples-Tests/SouthAmericanCurrenciesRESTfulControllerTest.class.st

```diff
@@ -145,9 +205,43 @@ SouthAmericanCurrenciesRESTfulControllerTest >> testGetArgentinePesoInSpanish [
 	| response request |
 	request := self
 		           requestToGETResourceIdentifiedBy: 'ARS'
-		           accepting:
-		           resourceController currencyVersion1dot0dot0MediaType.
-	request setAcceptLanguage: 'es'.
+		           representedAs:
+		           resourceController currencyVersion1dot0dot0MediaType
+						 in: 'es'.
+
+	response := resourceController
+		            currencyBasedOn: request
+		            within: self newHttpRequestContext.
+
+	self
+		assert: response isSuccess;
+		assert: response status equals: 200;
+		assert: response contentType asMediaType
+		equals: resourceController currencyVersion1dot0dot0MediaType;
+		withTheOnlyOneIn: response contentLanguageTags
+		do: [ :tag | self assert: tag equals: 'es-AR' asLanguageTag ];
+		assert: response varyHeaderNames includes: 'Accept-Language';
+		assert: response varyHeaderNames includes: 'Accept';
+		assert: response entityTag
+		equals: '"fd65007599711248b85d86917bd24c5aa29e610b"' asEntityTag.
+
+	self withJsonFromContentsIn: response do: [ :json | 
+		self
+			assert: json name equals: 'Peso';
+			assert: json symbol equals: '$';
+			assert: json isoCode equals: 'ARS';
+			assertThereAreNoLinksIn: json ]
+]
+
+{ #category : #tests }
+SouthAmericanCurrenciesRESTfulControllerTest >> testGetArgentinePesoInSpanishBasedOnLanguageQualityPriority [
+
+	| response request |
+	request := self
+		           requestToGETResourceIdentifiedBy: 'ARS'
+		           representedAs:
+		           resourceController currencyVersion1dot0dot0MediaType
+						 in: 'de,en;q=0.5,es;=0.6'.
```

Comment:
This is called a language priority list. See https://www.w3.org/TR/ltli/#dfn-language-range